litex/tools/litex_json2dts_zephyr.py

```diff
@@ -42,11 +43,31 @@ def ram_handler(name, parm, csr):
 
 
 def ethmac_handler(name, parm, csr):
-    dtsi  = dts_reg(" ".join([
-                hex(csr['csr_bases'][name]),
-                hex(parm['size']),
-                hex(csr['memories'][name]['base']),
-                hex(csr['memories'][name]['size'])]))
+    rx_registers = get_registers_of('ethmac_sram_writer', csr)
+    for reg in rx_registers:
+        reg['name'] = 'rx_' + reg['name']
+
+    tx_registers = get_registers_of('ethmac_sram_reader', csr)
+    for reg in tx_registers:
+        reg['name'] = 'tx_' + reg['name']
+
+    eth_buffers = {
+        'name': 'buffers',
+        'addr': csr['memories'][name]['base'],
+        'size': csr['memories'][name]['size'],
+        'type': csr['memories'][name]['type'],
+    }
+    registers = rx_registers + tx_registers + [eth_buffers]
+    if len(registers) == 0:
```

Comment:
we create `eth_buffers` and always include them in `registers`, so they can't be empty at this point, right?

Reply:
That is correct, removed this check, as `KeyError` would be raised during creation of `eth_buffers` anyway if there is no ethmac